### PR TITLE
fix: remove prop type warning and use only CSS for threshold dot

### PR DESCRIPTION
**Pull request by adamstankiewicz**
## Description

image

### Deploy Preview

Include a direct link to your changes in this PR's deploy preview here (e.g., a specific component page).

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: src/Popover/Popover.test.jsx [hunk 1]
```diff
@@ -6,17 +6,17 @@ import Popover from './index';
 describe('<Popover />', () => {
   describe('correct rendering', () => {
     it('renders with correct class for variant success', () => {
-      const wrapper = mount(<Popover variant="success" />);
+      const wrapper = mount(<Popover id="popover-id" variant="success" />);
```
Comment (adamstankiewicz) [inline src/Popover/Popover.test.jsx:9]: Missing required `id` props

File: src/ProgressBar/index.jsx [hunk 2]
```diff
@@ -116,8 +120,6 @@ ProgressBarAnnotated.propTypes = {
   variant: PropTypes.oneOf(VARIANTS),
   /** Specifies an additional `className` to add to the base element. */
   className: PropTypes.string,
-  /** Hide's the label visually. */
-  visuallyHidden: PropTypes.bool,
```
Comment (adamstankiewicz) [inline src/ProgressBar/index.jsx:120]: This prop doesn't do anything for the annotated progress bar, and is currently causing a prop type warning.

File: src/ProgressBar/index.jsx [hunk 3]
```diff
@@ -37,11 +37,21 @@ const ProgressBarAnnotated = ({
...
 
-  useEffect(() => {
+  const positionAnnotations = useCallback(() => {
     placeInfoAtZero(progressInfoRef, isProgressHintAfter, ANNOTATION_CLASS);
     placeInfoAtZero(thresholdInfoRef, isThresholdHintAfter, ANNOTATION_CLASS);
   }, [isProgressHintAfter, isThresholdHintAfter]);
 
+  useEffect(() => {
+    positionAnnotations();
+    const observer = new ResizeObserver(() => {
```
Comment (adamstankiewicz) [inline src/ProgressBar/index.jsx:47]: This helps keep the annotation info positioned more accurately when the progress bar itself changes widths (e.g., when the Paragon documentation site theme changes).

File: src/ProgressBar/index.jsx [hunk 4]
```diff
@@ -80,12 +90,6 @@ const ProgressBarAnnotated = ({
             srOnly
           />
         )}
-        {!!threshold && (
```
Comment (adamstankiewicz) [inline src/ProgressBar/index.jsx:83]: Used CSS only for this threshold dot instead of a DOM element.

**Review by long74100 — DISMISSED** [review]

**Review by long74100 — APPROVED** [review]

**Pull request merged**